src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_exception_thrown_before_slr_behavior.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.Config;
+    using Features;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class When_exception_thrown_before_slr_behavior : When_exception_thrown_during_retries
+    {
+        [Test]
+        public async void Outgoing_messages_should_not_be_delivered()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<FailingEndpoint>(b => b.When(bus => bus.SendLocalAsync(new FailingMessage())))
+                .WithEndpoint<ErrorSpy>()
+                .AllowSimulatedExceptions()
+                .Done(c => c.FailingMessageMovedToErrorQueueAndProcessedByErrorSpy)
+                .Run(TimeSpan.FromSeconds(20));
+
+            Assert.IsFalse(context.SideEffectMessageReceived, "When slr kicks in all messages pending to be sent should be dropped.");
+        }
+
+        public class FailingEndpoint : EndpointConfigurationBuilder
+        {
+            public FailingEndpoint()
+            {
+                EndpointSetup<DefaultServer>(
+                    b =>
+                    {
+                        var endpointName = AcceptanceTesting.Customization.Conventions.EndpointNamingConvention(typeof(ErrorSpy));
```

Comment:
It is a bit tricky and hidden to see the relationship to the `ErrorSpy` endpoint. Is there a more expressive way to write this test?